src/main/scala/io/iohk/ethereum/blockchain/sync/regular/StateNodeFetcher.scala

```diff
@@ -46,15 +47,13 @@ class StateNodeFetcher(
         requester
           .collect(stateNodeRequester => {
             val validatedNode = values
-              .asRight[String]
-              .ensure(s"Empty response from peer $peer, blacklisting")(_.nonEmpty)
-              .ensure("Fetched node state hash doesn't match requested one, blacklisting peer")(nodes =>
-                stateNodeRequester.hash == kec256(nodes.head)
-              )
+              .asRight[BlacklistReason]
+              .ensure(BlacklistReason.EmptyStateNodeResponse)(_.nonEmpty)
+              .ensure(BlacklistReason.WrongStateNodeResponse)(nodes => stateNodeRequester.hash == kec256(nodes.head))
 
             validatedNode match {
               case Left(err) =>
-                log.debug(err)
+                log.debug(err.description)
```

Comment:
while I enjoy the succinct form, maybe at least a bit of context in the message to help with grepping?

Reply:
wooops 😄